src/contexts/VideoCallContext.tsx

```diff
@@ -0,0 +1,243 @@
+// React + Web3 Essentials
+import React, { createContext, useEffect, useRef, useState } from 'react';
+
+// External Packages
+import Peer from 'simple-peer';
+import { io } from 'socket.io-client';
+
+// Internal Configs
+import { appConfig } from 'config';
+
+interface videoPayloadType {
+  userToCall: string;
+  fromUser: string;
+  signalData: any;
+  name: string;
+  status: number;
+}
+
+interface payloadType {
+  sender: string;
+  recipient: string;
+  identity: string;
+  source: string;
+}
+
+const VideoCallContext = createContext(null);
+
+// const socket = io('http://localhost:5000');
+// const socket = io('https://warm-wildwood-81069.herokuapp.com');
+
+const VideoCallContextProvider:React.FC<React.ReactNode> = ({ children }) => {
+  const [callAccepted, setCallAccepted] = useState<boolean>(false);
+  const [callEnded, setCallEnded] = useState<boolean>(false);
+  const [receiverPeerSignalled, setRecieverPeerSignalled] = useState<boolean>(false);
+  const [stream, setStream] = useState<MediaStream>();
+  const [name, setName] = useState<string>('Joe');
+  const [call, setCall] = useState<any>({});
+  const [me, setMe] = useState<string>('');
+
+  const myVideo = useRef();
+  const userVideo = useRef();
+  const connectionRef = useRef();
+
+  const initializeStream = async (address: string): Promise<void> => {
+    try {
+      const stream = await navigator.mediaDevices.getUserMedia({ video: true, audio: true });
+      setStream(stream);
+      myVideo.current.srcObject = stream;
```

Comment:
fix the error 'myVideo.current' is possibly 'undefined' in this line, also fix this in the entire file, there are quite a few places with typescript  error.

Reply:
ok...I will look into it

Reply:
Pushed changes...